server/src/main/proto/spine/server/catchup/catch_up.proto

```diff
@@ -0,0 +1,95 @@
+syntax = "proto3";
+
+package spine.server.catchup;
+
+import "spine/options.proto";
+import "spine/time_options.proto";
+
+option (type_url_prefix) = "type.spine.io";
+option java_package = "io.spine.server.delivery";
+option java_outer_classname = "CatchUpProto";
+option java_multiple_files = true;
+
+import "google/protobuf/any.proto";
+import "google/protobuf/timestamp.proto";
+
+import "spine/core/event.proto";
+
+// A state of a catch-up process.
+message CatchUp {
+    option (is).java_type = "CatchUpMixin";
+
+    CatchUpId id = 1;
+
+    // A description of a catch-up requested.
+    message Request {
+
+        // IDs of the projections under catch-up, packed as `Any`.
+        //
+        // If no particular targets are set, the catch-up is requested for all entities
+        // of the respective kind.
+        //
+        repeated google.protobuf.Any target = 1;
+
+        // Since which point in time the catch-up has been originally requested.
+        //
+        // This value is inclusive as for the timestamp of events to which this catch-up should
+        // be applicable.
+        //
+        google.protobuf.Timestamp since_when = 2 [(required) = true, (when).in = PAST];
+
+        // The type URLs of events to read from the Event Store.
+        repeated string event_type = 3;
+    }
+
+    // The original request.
+    Request request = 2 [(required) = true, (validate) = true];
+
+    // During the catch-up, this field holds the timestamp of the last event already processed.
+    google.protobuf.Timestamp when_last_read = 3;
+
+    // The current status of the catch-up process.
+    CatchUpStatus status = 4 [(required) = true];
+
+    // The index of the current catch-up round.
+    //
+    // In each round the batch of the events is read from the Event Store and posted
+    // to the respective `Inbox`es.
+    //
+    // The index of the first round is 0.
+    //
+    int32 current_round = 5;
+
+    // The total number of shards in the environment
+    // at the moment of the last dispatching of the historical messages.
+    //
+    int32 total_shards = 6;
+
+    // The indexes of the shards through which the historical messages were dispatched.
+    repeated int32 affected_shard = 7;
```

Comment:
Why not `ShardId`s?

Reply:
This is done to save up the space taken by this record in the storage with max BLOB size.  There may be lots of shards involved into the message dispatching, and Cloud Datastore [allows us to store `1,048,572` bytes per record at most](https://cloud.google.com/datastore/docs/concepts/limits). Basically, if there are 1000 shards,  we would take at least 4000 bytes more + the overhead for Proto message format, which I believe takes at least as  much.